vega_sim/builders/governance.py

```diff
@@ -44,14 +44,27 @@ def proposal_terms(
     update_volume_discount_program: Optional[
         vega_protos.governance.UpdateVolumeDiscountProgram
     ] = None,
-):
-    proposal_terms = vega_protos.governance.ProposalTerms(
-        closing_timestamp=int(closing_timestamp.timestamp()),
-        enactment_timestamp=int(enactment_timestamp.timestamp()),
-    )
-    if validation_timestamp is not None:
-        setattr(
-            proposal_terms, "validation_timestamp", validation_timestamp.timestamp()
+    for_batch_proposal: bool = False,
+) -> Union[
+    vega_protos.governance.ProposalTerms,
+    vega_protos.governance.BatchProposalTermsChange,
+]:
+    if not for_batch_proposal:
+        if closing_timestamp is None:
+            raise Exception(
+                "For non batch proposals changes closing timestamp must be set"
+            )
```

Comment:
Could we not have this check here?

I was thinking these build methods would be "dumb" so the fuzzers can try wrong things without errors being raised and us having to catch them else where (the exception being the `VegaProtoValueError`s raised by the wrapper).

Could have this sort of validation in any service methods that use the builder instead?